aten/src/ATen/native/UpSampleBilinear2d.cpp

```diff
@@ -6,158 +6,90 @@
 #include <ATen/native/UpSample.h>
 
 namespace at {
-namespace native {
-namespace {
+namespace meta {
 
-static void upsample_bilinear2d_out_cpu_template(
-    Tensor& output,
-    const Tensor& input,
-    IntArrayRef output_size,
-    bool align_corners,
-    c10::optional<double> scales_h,
-    c10::optional<double> scales_w) {
+TORCH_META_FUNC(upsample_bilinear2d) (
+  const Tensor& input, IntArrayRef output_size, bool align_corners, c10::optional<double> scales_h, c10::optional<double> scales_w
+) {
+  auto full_output_size = native::upsample_2d_common_check(input.sizes(), output_size);
+
+  // Allow for empty batch size but not other dimensions
   TORCH_CHECK(
-      output_size.size() == 2,
-      "It is expected output_size equals to 2, but got size ",
-      output_size.size());
-
-  int64_t output_height = output_size[0];
-  int64_t output_width = output_size[1];
-
-  int64_t nbatch = input.size(0);
-  int64_t channels = input.size(1);
-  int64_t input_height = input.size(2);
-  int64_t input_width = input.size(3);
-
-  upsample_2d_shape_check(
-      input,
-      Tensor(),
-      nbatch,
-      channels,
-      input_height,
-      input_width,
-      output_height,
-      output_width);
-
-  output.resize_({nbatch, channels, output_height, output_width}, input.suggest_memory_format());
```

Comment:
Looks like we don't suggest `memory_format` here either, the same as in https://github.com/pytorch/pytorch/pull/52065/files#r589557711

cc @vfdev-5